src/ui/components/OverlayKeyBindings/component.ts

```diff
@@ -0,0 +1,87 @@
+import Component, { tracked } from '@glimmer/component';
+import { ESCAPE, TAB } from '../../../utils/keys';
+
+const FOCUSABLE_ELEMENTS = [
+  '[tabindex]',
+  'a',
+  'button:not([disabled])',
+  'input:not([disabled])',
+  'select:not([disabled])',
+  'textarea:not([disabled])'
+];
+
+const handleKeydown = function(e) {
+  if (e.keyCode === ESCAPE) {
+    this.escapeKeyPressed();
+  }
+
+  if(e.keyCode === TAB) {
+    this.tabKeyPressed(e);
+  }
+};
+
+const keepFocus = function(e) {
+  if(!this.el.contains(e.target)) {
+    this.focusableElements[0].focus();
+  }
+};
+
+export default class extends Component {
+  handleKeydown = null;
+  keepFocus = null;
+  el = null;
+
+  constructor(options) {
+    super(options);
+    this.handleKeydown = handleKeydown.bind(this);
+    this.keepFocus = keepFocus.bind(this);
+  }
+
+  didInsertElement() {
+    let firstNode = <HTMLElement>this.bounds.firstNode;
+    this.el = firstNode.nodeType === 3 ? firstNode.nextElementSibling : firstNode;
+
+    let firstFocusable = <HTMLElement>this.focusableElements[0];
+    firstFocusable.focus();
+
+    document.addEventListener('keydown', this.handleKeydown);
+    document.body.addEventListener('focus', this.keepFocus, true);
+  }
+
+  willDestroy() {
+    this.teardown();
+  }
+
+  teardown() {
```

Comment:
I would like to just have this functionality in the `willDestroy` hook but this method is part of the battle with event listeners and the setTimeout in `src/ui/components/Main/component.ts`

Reply:
Maybe the solution is a service than manages keyboard events... 💭